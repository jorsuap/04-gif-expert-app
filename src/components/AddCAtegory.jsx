
import { useState } from "react";
import PropTypes from "prop-types";

export const AddCAtegory = ( { onNewCategory } ) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue( target.value )
    };

    const onSubmit = ( event )=> {
        event.preventDefault();
        if (inputValue.trim().length <= 1) {
            return;
        }
        onNewCategory( inputValue.trim() );
        setInputValue('');
    };

  return (
    <form onSubmit = { onSubmit }>
        <input
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}

// AddCAtegory.propTypes = {
//     onNewCategory: PropTypes.onNewCategory
// }


