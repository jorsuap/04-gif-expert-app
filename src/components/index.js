export * from './AddCAtegory';
export * from './GifGrid';
export * from './GifCard';