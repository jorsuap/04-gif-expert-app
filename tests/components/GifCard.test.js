
import { render } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";
import { screen } from "@testing-library/react";


describe('GifCard', ()=> {
    const title = 'Goku';
    const url = 'https://chatgpt.com/';

    test('Snapshot', ()=> {
        const { container }  = render( <GifCard title={title} url={url}/>);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con url y alt', ()=>{
        render( <GifCard title={title} url={url}/>);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('Title', (()=>{
        render( <GifCard title={ title } url={ url }/>);

        expect( screen.getByText( title )).toBeTruthy();
    }))
})