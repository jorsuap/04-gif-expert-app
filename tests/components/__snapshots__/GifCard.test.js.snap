// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://chatgpt.com/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
